tests/dspa_test.go

```diff
@@ -0,0 +1,63 @@
+//go:build test_integration
+
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	systemsTesttUtil "github.com/opendatahub-io/data-science-pipelines-operator/tests/util"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("A successfully deployed DSPA", func() {
+
+	podCount := 5
+
+	Context("with default MariaDB and Minio", func() {
+		It(fmt.Sprintf("should have %d pods", podCount), func() {
+			podList := &corev1.PodList{}
+			listOpts := []client.ListOption{
+				client.InNamespace(DSPANamespace),
+			}
+			err := clientmgr.k8sClient.List(ctx, podList, listOpts...)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(podList.Items)).Should(Equal(podCount))
+		})
+
+		It(fmt.Sprintf("should have a ready %s deployment", "DSP API Server"), func() {
```

Comment:
can we convert this into a table-driven loop since the input and check for each set of statements is essentially identical except for the input and output strings

Reply:
I'm not even sure if we want to keep the structure as is (or these specific test), these are first pass test the focus is more on the suite overall, If we're happy with the set up, what you are proposing sounds like a good-first-issue

Reply:
ok opened https://github.com/opendatahub-io/data-science-pipelines-operator/issues/421 in case we want to do this in the future.  FWIW the suggested design pattern is common across golang projects so we should be trying to leverage it when writing new tests as well.